core/sds.c

```diff
@@ -1318,16 +1380,20 @@ oio_sds_upload_prepare (struct oio_sds_ul_s *ul, size_t size)
 
 	EXTRA_ASSERT(!ul->hexid || oio_str_ishexa1(ul->hexid));
 
-	GSList *_chunks = NULL;
-	/* Parse the output, as a JSON array of objects with fields
-	 * depicting chunks */
+	GSList *_chunks = NULL;  // GSList<struct chunk_s*>
+	/* Parse the output, as a (JSON object with a) JSON array of objects
+	 * with fields depicting chunks */
 	if (!err) {
 		struct json_tokener *tok = json_tokener_new ();
 		struct json_object *jbody = json_tokener_parse_ex (tok,
 				reply_body->str, reply_body->len);
-		if (!json_object_is_type(jbody, json_type_array))
+		if (json_object_is_type(jbody, json_type_array))
+			err = _chunks_load(&_chunks, jbody);
+		else if (json_object_is_type(jbody, json_type_object))
```

Comment:
You should reverse with the first condition. After the update, there will be only JSON objects.